pkg/ruler/mapper.go

```diff
@@ -59,8 +63,16 @@ func (m *mapper) MapRules(user string, ruleConfigs map[string][]legacy_rulefmt.R
 	}
 
 	for _, existingFile := range existingFiles {
-		fullFileName := path + existingFile.Name()
-		ruleGroups := ruleConfigs[existingFile.Name()]
+		fullFileName := filepath.Join(path, existingFile.Name())
+
+		// Ensure the namespace is decoded from base64 to see if it is still required
+		decodedNamespace, err := base64.URLEncoding.DecodeString(existingFile.Name())
+		if err != nil {
+			level.Warn(m.logger).Log("msg", "unable to remove rule file on disk", "file", existingFile.Name(), "err", err)
```

Comment:
```suggestion
			level.Warn(m.logger).Log("msg", "unable to remove rule file on disk", "file", fullFileName.Name(), "err", err)
```

Reply:
These aren't rule files meant for the storage and persistence of rules. They are temporary files that are loaded by the upstream prometheus manager code for evaluation. They are essentially stateless so no upgrade path is required.